scanner/src/main/java/edu/ucr/cs/riple/scanner/Config.java

```diff
@@ -53,6 +56,26 @@ public interface Config {
   @Nonnull
   Path getOutputDirectory();
 
+  Context getContext();
+
+  class Context {
```

Comment:
I like the general idea here, which is that we are passing our own "VisitorState" of sorts. That said, it feels a bit strange to be passing the "context" inside the "configuration", rather than the other way around (the configuration being part of the context in which we are running). How massive of a refactor would be to have an `ScannerContext` top level object that gets passed around and has `getConfig()` in addition to `getNextMethodId()` (let's not use `getNextUniqueID()` unless we plan to use it across different types of objects)

Reply:
My suggestion to rename it to `getNextMethodId()` depends on whether the ID is used to number things other than methods.

The method I think should be removed is not this, but `edu.ucr.cs.riple.core.Config.getNextModuleUniqueID()` (As a test, I am pretty sure if you make that one private, nothing is calling it outside of `edu.ucr.cs.riple.core.Config` :/

Reply:
@lazaroclapp Thank you for the clarification, I mistakenly thought you are talking about `getNextUniqueID` in `Scanner` config. Changed the visibility from `public` to `private`. 26083ada5e91d7ef4b8cc3ab3bbe994c51c68599

Reply:
0b025195ebd9f4a8c53c69ad3ea34db3e77cf181